RockLib.Messaging/HeaderDictionary.cs

```diff
@@ -62,27 +63,15 @@ public bool TryGetValue<T>(string key, out T value)
                 var converter = TypeDescriptor.GetConverter(typeof(T));
                 if (converter.CanConvertFrom(objectValue.GetType()))
                 {
-                    try
-                    {
-                        value = (T)converter.ConvertFrom(objectValue);
-                        return true;
-                    }
-                    catch
-                    {
-                    }
+                    value = (T)converter.ConvertFrom(objectValue)!;
```

Comment:
Should we add another documentation somewhere to denote that this now could throw an exception?

Reply:
In this case, there's already a check to see if that object's type can be converted, so there wasn't a reason to have a `try/catch` there in the first place.